drivers/media/platform/bcm2835/bcm2835-unicam.c

```diff
@@ -1012,6 +1113,106 @@ static int unicam_s_fmt_vid_cap(struct file *file, void *priv,
 	return 0;
 }
 
+static int unicam_enum_fmt_meta_cap(struct file *file, void *priv,
+				    struct v4l2_fmtdesc *f)
+{
+	struct unicam_node *node = video_drvdata(file);
+	struct unicam_device *dev = node->dev;
+	struct v4l2_subdev_mbus_code_enum mbus_code;
+	const struct unicam_fmt *fmt = NULL;
+	int ret = 0;
+
+	if (node->pad_id == IMAGE_PAD || f->index != 0)
```

Comment:
Has something else range checked ```pad_id``` to ensure it isn't greater than the number of pads? If not then why not switch to ```pad_id != METADATA_PAD``` and kill 2 birds with one stone?

Reply:
Done.

Should not matter I expect.  The pad_id field is only ever set by us in probe() so the values are explicit unless something has gone horribly wrong.